internal/pkg/manifest/svc.go

```diff
@@ -342,54 +130,6 @@ func (a *Autoscaling) IsEmpty() bool {
 		a.Requests == nil && a.ResponseTime == nil
 }
 
-// ServiceProps contains properties for creating a new service manifest.
-type ServiceProps struct {
-	Name       string
-	Dockerfile string
-}
-
-// UnmarshalService deserializes the YAML input stream into a service manifest object.
-// If an error occurs during deserialization, then returns the error.
-// If the service type in the manifest is invalid, then returns an ErrInvalidManifestType.
-func UnmarshalService(in []byte) (interface{}, error) {
-	am := Service{}
-	if err := yaml.Unmarshal(in, &am); err != nil {
-		return nil, fmt.Errorf("unmarshal to service manifest: %w", err)
-	}
-	typeVal := aws.StringValue(am.Type)
-
-	switch typeVal {
-	case LoadBalancedWebServiceType:
-		m := newDefaultLoadBalancedWebService()
-		if err := yaml.Unmarshal(in, m); err != nil {
-			return nil, fmt.Errorf("unmarshal to load balanced web service: %w", err)
-		}
-		return m, nil
-	case BackendServiceType:
-		m := newDefaultBackendService()
-		if err := yaml.Unmarshal(in, m); err != nil {
-			return nil, fmt.Errorf("unmarshal to backend service: %w", err)
-		}
-		if m.BackendServiceConfig.Image.HealthCheck != nil {
-			// Make sure that unset fields in the healthcheck gets a default value.
-			m.BackendServiceConfig.Image.HealthCheck.applyIfNotSet(newDefaultContainerHealthCheck())
-		}
-		return m, nil
-	case ScheduledJobType:
-		m := newDefaultScheduledJob()
-		if err := yaml.Unmarshal(in, m); err != nil {
-			return nil, fmt.Errorf("unmarshal to scheduled job: %w", err)
-		}
-		return m, nil
-	default:
-		return nil, &ErrInvalidSvcManifestType{Type: typeVal}
-	}
-}
-
-func durationp(v time.Duration) *time.Duration {
-	return &v
-}
-
```

Comment:
parsePortMapping is used for sidecar port parsing. Maybe we want to put it in `workload.go`.

Reply:
Great idea!